sound/soc/sof/intel/icl.c

```diff
@@ -106,6 +107,10 @@ const struct snd_sof_dsp_ops sof_icl_ops = {
 	.trace_release = hda_dsp_trace_release,
 	.trace_trigger = hda_dsp_trace_trigger,
 
+	/* client ops */
+	.ipc_register_clients = icl_register_clients,
+	.ipc_unregister_clients = icl_unregister_clients,
```

Comment:
The `ipc` prefix was suggested to make it explicit that these clients are for features implemented in SOF firmware and can be accessed via IPC messages.
This is the case with the probes as well. For HDA we need to do some configuration on the host side apart from pure SOF feature because of how the platform is designed.
In theory I think probes could be implemented for non HDA platforms, but for now we only support HDA, the callbacks are mandatory atm.

Yes, that is one of the reason to have separated function for client registration and we will have other clients int he future (I'm looking at the dma trace atm).

Reply:
hm, to me it sounds a bit like a "client registration method from the IPC subsystem." Maybe `ipc_client_register()`? Or even a bold `dsp_client_register()`? I'm not good at inventing names so feel free to disagree, just something that I couldn't understand from the names themselves

Reply:
@lyakh, I would go with `ipc_client_register()` or rather `register_ipc_clients()` ?
The callback is used by the core to call in to the platform code and register platform dependent IPC clients. Atm we have only the probes (which is only available for HDA), but I'm planning to convert the (dma-)trace - which  is again HDA dependent by the looks. Then the audio en large.